### PR TITLE
Migrate to ethers v5

### PR DESCRIPTION
**Pull request by TomAFrench**
Companion PR to https://github.com/balancer-labs/balancer-exchange/pull/127

I've updated all references to ethersv4 to point at ethersv5 and made the edit to the multicall abi as mentioned in the other PR.

~~It's quite possible that https://github.com/balancer-labs/pool-management may need to be updated in parallel (I haven't looked into this yet.) but possibly not as it seems to be a case of v5 just being stricter than v4 so a v4 project may not have any troubles.~~ pool-managment doesn't use this package

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,8 +36,13 @@
     },
     "dependencies": {
         "bignumber.js": "^9.0.0",
-        "ethers": "^4.0.39",
         "isomorphic-fetch": "^2.2.1",
         "typescript": "^3.8.3"
+    },
+    "peerDependencies": {
```
Comment (bonustrack) [inline package.json:42]: The script `build` will fail for me with the message
```
error TS2307: Cannot find module '@ethersproject/abi'.
```
If the ethers dependencies are not in `dependencies`. Do you also have the same error?

Comment (TomAFrench) [reply package.json:47]: Ah sorry, I forgot to add it as a dev dependency. It should be able to be built while using the user's installation of ethers now.

**Comment by bonustrack** [comment]
@TomAFrench can you PR against master?

**Comment by TomAFrench** [comment]
Ah sorry, I merged master into develop to get #38 in here. Would you like me to undo that or is this fine?

**Pull request merged**